symengine/sets.cpp

```diff
@@ -620,6 +632,107 @@ RCP<const Set> Complement::set_complement(const RCP<const Set> &o) const
     return container_->set_complement(newuniv);
 }
 
+ConditionSet::ConditionSet(const RCP<const Basic> sym,
+                           RCP<const Boolean> condition)
+    : sym(sym), condition_(condition)
+{
+    SYMENGINE_ASSIGN_TYPEID()
+    SYMENGINE_ASSERT(ConditionSet::is_canonical(sym, condition))
+}
+
+bool ConditionSet::is_canonical(const RCP<const Basic> sym,
```

Comment:
I think it should remain un-evaluated as SymPy does.
Because if in any case, `x` gets dependent on `y`, we can't be sure that it is universal set or empty set (it can be an interval as well, in your example put `y=x+5`)
And if we consider that `x` and `y` are always going to be independent, which can easily be discarded by using three inter-dependent variables in reasoning.